include/mls/crypto.h

```diff
@@ -102,83 +79,65 @@ struct HPKEPublicKey
 {
   bytes data;
 
-  HPKEPublicKey() = default;
-  HPKEPublicKey(bytes data);
-
   HPKECiphertext encrypt(CipherSuite suite,
                          const bytes& aad,
                          const bytes& pt) const;
-  bytes to_bytes() const;
 
   TLS_SERIALIZABLE(data)
   TLS_TRAITS(tls::vector<2>)
 };
 
-class HPKEPrivateKey
+struct HPKEPrivateKey
 {
-public:
   static HPKEPrivateKey generate(CipherSuite suite);
   static HPKEPrivateKey parse(CipherSuite suite, const bytes& data);
   static HPKEPrivateKey derive(CipherSuite suite, const bytes& secret);
 
+  HPKEPrivateKey() = default;
+
+  bytes data;
+  HPKEPublicKey public_key;
+
   bytes decrypt(CipherSuite suite,
                 const bytes& aad,
                 const HPKECiphertext& ct) const;
-  HPKEPublicKey public_key() const;
 
-  TLS_SERIALIZABLE(_data, _pub_data)
-  TLS_TRAITS(tls::vector<2>, tls::vector<2>)
+  TLS_SERIALIZABLE(data, public_key)
+  TLS_TRAITS(tls::vector<2>, tls::pass)
 
 private:
-  bytes _data;
-  bytes _pub_data;
-
-  HPKEPrivateKey(CipherSuite suite, bytes data);
+  HPKEPrivateKey(bytes priv_data, bytes pub_data);
 };
 
 // Signature Keys
-class SignaturePublicKey
+struct SignaturePublicKey
 {
-public:
-  SignaturePublicKey();
-  SignaturePublicKey(CipherSuite suite, bytes data);
+  bytes data;
 
-  void set_cipher_suite(CipherSuite suite);
-  void set_signature_scheme(SignatureScheme scheme);
-  SignatureScheme signature_scheme() const;
-  bool verify(const bytes& message, const bytes& signature) const;
-  bytes to_bytes() const;
+  bool verify(const CipherSuite& suite,
+              const bytes& message,
+              const bytes& signature) const;
 
```

Comment:
would be nice to have a version that takes in jus the scheme (and no suite) .. esp it helps for X509 work where i did add an API in that effect ..

Reply:
Nope.  SignatureScheme is no more!